# Q1 Cutlets

- H0 = There is no significant difference in the diameter of the cutlet betwwen two units.
- HA = There is significant difference in the diameter of the cutlet betwwen two units.

In [1]:
import pandas as pd
from scipy import stats

In [2]:
Cutlets = pd.read_csv("D:\Assignment\Hypothesis Testing\Cutlets.csv")

In [3]:
Cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [4]:
stats.ttest_ind(Cutlets['Unit A'], Cutlets['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

- P- Value = 0.48
- With 5% significance level the Alpha value = 0.05
- P- Value > Alpha ( 0.48 > 0.05)
- Fail to reject Null Hypothesis [H0]

There is 'no' significant difference in the diameter of the cutlet between two units

---------------------------------------------------------------------------------------------------------------------------

# Q2 LabTAT

- H0 = No difference in average TAT among the different laboratoraries.
- HA = Difference in average TAT among the different laboratoraries.

In [5]:
LabTAT = pd.read_csv("D:\Assignment\Hypothesis Testing\LabTAT.csv")

In [6]:
LabTAT.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [7]:
stats.f_oneway(LabTAT.iloc[:,0],LabTAT.iloc[:,1],LabTAT.iloc[:,2],LabTAT.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

- P- Value ~ 0
- With 5% significance level the Alpha value = 0.05
- P- value < Alpha (~0 < 0.05)
- Reject Null Hypothesis [H0]

There is difference in average TAT among the different laboratoraries.

---------------------------------------------------------------------------------------------------------------------------

# Q3 Buyer_Ratio

- H0 = All proportion are equal
- HA = Not all proportions are equal

In [8]:
import numpy as np
from scipy.stats import chi2_contingency

In [9]:
Buyer_Ratio = pd.read_csv("D:\Assignment\Hypothesis Testing\BuyerRatio.csv")

In [10]:
Buyer_Ratio.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [11]:
Buyer_Ratio

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [12]:
Observed_Value = np.array([[50,142,131,70], [435,1523,1356,750]])

In [13]:
p_value = chi2_contingency(Observed_Value) # o/p is (Chi2 stats value, p_value, df, expected obsvations)
p_value

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [14]:
p_value[1]

0.6603094907091882

- P- Value = 0.660
- With 5% significance level the Alpha value = 0.05
- P- value > Alpha (0.660 > 0.05)
- Fail to reject Null Hypothesis [H0]

Not all proportions are equal

---------------------------------------------------------------------------------------------------------------------------

# Q4 CustomerOrderForm

- H0 = No defective varies by centre
- HA = Defective varies by centre

In [15]:
CustomerOrderForm = pd.read_csv("D:\Assignment\Hypothesis Testing\Costomer+OrderForm.csv")

In [16]:
CustomerOrderForm.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [17]:
CustomerOrderForm.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [18]:
CustomerOrderForm.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [19]:
CustomerOrderForm.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [20]:
CustomerOrderForm.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [21]:
CustomerOrderForm.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [22]:
Observed_Value = np.array([
    [271, 267, 269, 280],
    [29, 33, 31, 20]
])

In [23]:
p_value  = chi2_contingency(Observed_Value) # o/p is (Chi2 stats value, p_value, df, expected obsvations)
p_value

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [24]:
p_value[1]

0.2771020991233135

- P- Value = 0.277
- With 5% significance level the Alpha value = 0.05
- P- value > Alpha (0.277 > 0.05)
- Fail to reject Null Hypothesis [H0]

Defective varies by centre